internal/events/webhooks/webhooks.go

```diff
@@ -224,7 +225,18 @@ func (wh *WebHooks) attemptRequest(sub *core.Subscription, event *core.EventDeli
 		}
 	}
 
-	req, err = wh.buildRequest(sub.Options.TransportOptions(), firstData)
+	// Taking a copy of the global plugin client because
+	// 1) We do not want to modify that global instance
+	// 2) We want to keep the global configuration for webhooks
+	httpClient := *wh.client.GetClient()
```

Comment:
Are we sure that this is a safe copy?
Feels a little dangerous to be assuming that the changes we're making are not modified sub-structures of `resty` (as it exists now or in the future) that are pointer referenced.

The `ffresty` object has the ability to build a config, which is from the Viper config that goes into a Go structure.
It might be safer to construct the client from that with the TLS options modified, or each subscription.

Reply:
I originally went the second way but it meant dealing with config merges which I thought would be more complex. The ffresty config section comes from the `events.webhooks` root section but we have the TLS Config section under `namespaces.predefined[].tlsConfigs[].tls but I guess as you say I could store the `config.Section` the Webhook object create a new resty client and then call `newClient.SetTLSClientConfig`


I agree that it can be dangerous and the deep copy is not guaranteed. I couldn't find anywhere where someone takes a deep copy of an http client

Reply:
> we could optimise this and store the client in the subscription object

That would definitely be an option to change in this code, that would let me feel comfortable to merge.
e.g. we construct from config uniquely for each subscription, and don't try to do any object copying.

The pattern I was referring to is here:
https://github.com/hyperledger/firefly-common/blob/86d110a4e827b03aeca0723218723c52f70e5574/pkg/wsclient/wsconfig.go#L59-L82

I had it in my head that we'd done it for `ffresty` as well as WebSockets, but it seems not.